snmp/datadog_checks/snmp/snmp.py

```diff
@@ -101,11 +101,17 @@ def _get_profiles_mapping(self):
         """
         Get the mapping from sysObjectID to profile.
         """
-        profiles_by_oid = {}
+        profiles_by_oid = {}  # type: Dict[str, str]
         for name, profile in self.profiles.items():
             sys_object_oid = profile['definition'].get('sysobjectid')
             if sys_object_oid is not None:
-                profiles_by_oid[sys_object_oid] = name
+                profile_matches = [prof_name for oid, prof_name in profiles_by_oid.items() if oid == sys_object_oid]
```

Comment:
I may miss something, but can't you just check that `sys_object_oid`isn't in the dict and raise an error if that's the case?

Reply:
Yes, you are correct. I previously thought `profiles_by_oid` was structured differently so that method was a remnant of it. It's much cleaner this way!